### ISSUE TITLE
when export model file，report an error

### ISSUE THREAD

**Issue by jobsjiang** [ultralytics/yolov5#9366]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

custom model traing，Classes = 2，when export model file with --train report an error， if not with --train no error
```
Traceback (most recent call last):
  File "export.py", line 595, in <module>
    main(opt)
  File "export.py", line 590, in main
    run(**vars(opt))
  File "/root/anaconda3/envs/yolov5/lib/python3.8/site-packages/torch/autograd/grad_mode.py", line 27, in decorate_context
    return func(*args, **kwargs)
  File "export.py", line 499, in run
    shape = tuple((y[0] if isinstance(y, tuple) else y).shape)  # model output shape
AttributeError: 'list' object has no attribute 'shape'
```

**Comment by glenn-jocher** [comment]
@jobsjiang I'm able to reproduce this with `python export.py --train`

<img width="827" alt="Screenshot 2022-09-11 at 13 58 23" src="https://user-images.githubusercontent.com/26833433/189523840-3a326639-9cf5-41d0-8bac-2c5ea2adfcfa.png">

I was thinking of deprecating this setting as normal workflows don't involve training on exported models. What is your use case or workflow? Thanks!

**Comment by jobsjiang** [comment]
> @jobsjiang I'm able to reproduce this with `python export.py --train`
> 
> <img alt="Screenshot 2022-09-11 at 13 58 23" width="827" src="https://user-images.githubusercontent.com/26833433/189523840-3a326639-9cf5-41d0-8bac-2c5ea2adfcfa.png">
> 
> I was thinking of deprecating this setting as normal workflows don't involve training on exported models. What is your use case or workflow? Thanks!

I just convert list to array to avoid the error，now finding the reason of the error

**Comment by glenn-jocher** [comment]
@jobsjiang good news 😃! Your original issue may be fixed ✅ in PR #9429. This PR removes train mode exports, which have no common use cases and are not currently working in master.

To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!
